architecture.md

```diff
@@ -0,0 +1,122 @@
+#  Verifiable Credential Issuer HTTP API Architecture Model
+This document describes the architecture model and context for the Verifiable
+Credential Issuer HTTP API.
+
+## Overview
+The Verifiable Credential Issuer HTTP API is designed to be a consistent API for
+Issuer Services to call in order to generate Verifiable Credentials. This API is
+intended for Issuer Services who are able to manage and secure external
+connections to Holders. This API encapsulates and abstracts the Verifiable
+Credential data model, format, and proof methods from Issuer Services. This API
+also abstracts any underlying DID methods, resolver capabilities, and
+cryptographic algorithms from the Issuer Services.
+
+## Objectives
+The following capture the objectives of this API.
+1. Provide an open and consistent HTTP based API for Issuer Services to leverage
+when generating Verifiable Credentials.
+1. Abstract the underlying data model, formats and proof mechanisms from the
```

Comment:
This item is believed to be in conflict with https://github.com/w3c-ccg/vc-issuer-http-api/pull/18#issuecomment-595202024 which is problematic. I don't see it that way, but seems @llorllale does. Fundamentally, we shouldn't be "abstracting" the underlying data model... because the data model *is* the Verifiable Credentials Data Model... and we need to deliver it via *some* representation of the data model.

Reply:
I don't think this line necessarily requires us to implement an API that is more complex than a simple issuer... consider that both VC-JWT and LD-Proofs can be constructed from the same json object... I think having this diagram merged could help us continue to refine the potential ways of abstracting...

Reply:
My fault for being super vague on this point - presumably an API abstracts _something_, so I put in some assumptions.

Reply:
@msporny

> This item is believed to be in conflict with [#18 (comment)](https://github.com/w3c-ccg/vc-issuer-http-api/pull/18#issuecomment-595202024) which is problematic. I don't see it that way, but seems @llorllale does. Fundamentally, we shouldn't be "abstracting" the underlying data model... because the data model _is_ the Verifiable Credentials Data Model... and we need to deliver it via _some_ representation of the data model.
> 
> If this item is going to be used as an argument against a simple API, we should strike it.

We don't see it that way but we are certainly open for discussion.